site/content/blog/2022-09-10-zap-extender-scripts/index.md

```diff
@@ -0,0 +1,140 @@
+---
+title: "Hacking ZAP - ZAP Extender Scripts"
+description: "Use ZAP as a web server, subscribe to internal ZAP events, and more!"
+summary: >
+    An overview of ZAP Extender scripts with examples.
+    Use ZAP as a web server, subscribe to internal ZAP events, and more!
+images:
+- https://www.zaproxy.org/blog/2022-09-10-zap-extender-scripts/images/_zapbot-extender-script.png
+type: post
+tags:
+- hacking
+- scripts
+- youdontknowzap
+date: "2022-09-10"
+addSocialPreview: true
+authors:
+- akshath
+---
+
+
+I recently learnt about ZAP Extender scripts and I wish I knew about them sooner!
+
+ZAP Extender Scripts allow dynamically adding new functionality to ZAP while it is running.
+This can include graphical elements and new API endpoints.
+
+The `install` function in Extender scripts is called when the script is enabled, and the `uninstall` function is called when it is disabled.
+
+This blog post contains some example ZAP Extender scripts that can do some neat tricks.
+These are just examples, and how you use ZAP scripts depends on how creative you can be!
+
+# Use ZAP as a web server
+While ZAP is intended to be used as a forward proxy, it can also act as an HTTP web server!
+This could be useful when you're already inside ZAP and need a mock target to test your new scan rule.
+Here's a ZAP Extender script that starts an echo server on port 9000 with the path "/echo":
```

Comment:
Should be mentioned that the examples will only work on newer ZAP versions (weekly and > 2.11).